packages/ui/nodes/cloud/src/CloudAttachmentElement/CloudAttachmentElement.styles.ts

```diff
@@ -0,0 +1,38 @@
+import { Value } from '@udecode/plate-core';
+import { createStyles } from '@udecode/plate-styled-components';
+import { css } from 'styled-components';
+import tw from 'twin.macro';
+import { statusBarStyleValues } from '../StatusBar/StatusBar.styles';
+import { CloudAttachmentElementStyleProps } from './CloudAttachmentElement.types';
+
+export const getCloudAttachmentElementStyles = <V extends Value>(
+  props: CloudAttachmentElementStyleProps<V>
+) => {
+  const { focused, selected } = props;
+  return createStyles(
+    { prefixClassNames: 'CloudAttachmentElement', ...props },
+    {
+      // container
+      root: [
+        tw`relative flex items-center h-10 max-w-sm gap-2 p-4 my-4 bg-black bg-white border border-gray-200 border-solid rounded-lg`,
+        focused &&
+          selected &&
+          tw`border-blue-400` &&
+          css`
+            box-shadow: 0 0 1px 3px #60a5fa;
+          `,
+      ],
+      // attachment icon
+      attachmentContainer: [tw`flex-shrink-0 text-gray-400`],
+      attachmentIcon: [],
+      // body
+      bodyContainer: [tw`flex-grow`],
+      bodyFilename: [tw`text-base`],
+      bodyCaption: [tw`text-sm text-gray-500`],
+      // download
+      downloadContainer: [tw`flex-shrink-0 w-8 h-8 ml-4 duration-200`],
+      downloadIcon: [tw`text-gray-400 cursor-pointer hover:text-blue-500`],
+      ...statusBarStyleValues,
+    }
+  );
+};
```

Comment:
For new features, we should directly adopt the new UI pattern, see the roadmap as reference https://github.com/orgs/udecode/projects/3

It consists of not using `styled-components` and exporting unstyled components from the non-ui package. The ui package only includes a (dumb) styled component. See the image plugin as reference.

Reply:
I may be in a little over my head on this one.

The cloud plugin has a lot of nested elements and it also has a shared element which also has nested elements (the \<StatusBar />). I'm uncertain how I should be modeling them and I think, to some degree, these decisions haven't been made yet based on my quick perusal of the components that you have upgraded already to the new UI pattern.

I wonder if maybe this is okay to punt to a later version when more of the use/edge cases have been worked out.

What do you think?

Reply:
Sure, let's do that later.